sdk/storage/MigrationGuide.md

```diff
@@ -0,0 +1,477 @@
+UploadBlockBlobOptions uploadOptions;
+uploadOptions.HttpHeaders.ContentHash.Algorithm = HashAlgorithm::Md5;
+uploadOptions.HttpHeaders.ContentHash.Value = precalculatedContentHash;
+blobClient.Upload(stream, uploadOptions);
+
+// download whole blob and get stored content hash property
+auto response = blobClient.Download();
+auto hashAlgorithm = response.Value.Details.HttpHeaders.ContentHash.Algorithm;  // This is always MD5
+auto md5 = response.Value.Details.HttpHeaders.ContentHash.Value;
+
+// validate stream against hash in your workflow
+```
+
+#### Transactional MD5 and CRC64
+
+Transactional hashes are not stored and have a lifespan of the request they are calculated for. Transactional hashes are verified by the service on upload.
+
+v7.5 provided transactional hashing on uploads and downloads through opt-in request options. MD5 and Storage's custom CRC64 were supported. The SDK calculated and validated these hashes automatically when enabled. The calculation worked on any upload or download method.
+
+v7.5
+
+```C++
+blob_request_options options;
+options.set_use_transactional_md5(false);  // true to use MD5 on all blob content transactions.
+options.set_use_transactional_crc64(false);  // true to use CRC64 on all blob content transactions.
+```
+
+v12 does not currently provide this functionality. Users who manage their own individual upload and download HTTP requests can provide a precalculated MD5 on upload and access the MD5 in the response object. v12 currently offers no API to request a transactional CRC64.
+
+```C++
+// upload a block with transactional hash calculated by user
+StageBlockOptions stageBlockOptions;
+stageBlockOptions.TransactionalContentHash = ContentHash();
+stageBlockOptions.TransactionalContentHash.Algorithm = HashAlgorithm::Md5;  // HashAlgorithm::Crc64 to use CRC64.
+stageBlockOptions.TransactionalContentHash.Value = precalculatedContentHash;
+blobClient.StageBlock(blockId, blockContentStream, stageBlockOptions);
+
+// upload more blocks as needed
+
+// commit block list
+blobClient.CommitBlockList(blockList);
+
+// download any range of blob with transactional checksum requested (maximum 4 MB for downloads)
+DownloadBlobOptions downloadOptions;
+downloadOptions.RangeHashAlgorithm = HashAlgorithm::Md5;  // HashAlgorithm::Crc64 to use CRC64.
+auto response = blobClient.Download();
+auto hashAlgorithm = response.Value.Details.HttpHeaders.ContentHash.Algorithm;
+auto hashValue = response.Value.Details.HttpHeaders.ContentHash.Value;
+// validate stream against hash in your workflow
+```
+
+### Resiliency
+
+#### Retry policy
+
+v7.5
+```C++
+blob_request_options options;
+options.set_retry_policy(exponential_retry_policy(delta_backoff, max_attempts));
+```
+
+v12
+```C++
+Blobs::BlobClientOptions options;
+// The only supported mode is exponential.
+options.Retry.RetryDelay = std::chrono::milliseconds(delta_backoff);
+options.Retry.MaxRetries = maxAttempts;
+```
+
+## Additional information
```

Comment:
How to construct a SAS v1 vs v12?
How to specify HTTP client options v1 vs v12? e.g. Proxy/CA_CERT location/Disable cert revokation list etc.
How to abort/cancel a request or specify a client timeout?
I'm OK with those info on this page or referencing samples.

Reply:
Do you want me to cover everything in this document?

Reply:
We need someplace to cover those. Feel free to cover it in other places you think appropriate.

Reply:
We have a [sample](https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/storage/azure-storage-blobs/sample/blob_sas.cpp) for SAS generation, the link of which is at the bottom of this guide.

I might able to accept to add timeout in this guide, but I want to hold it until we have get requests from customers.

I won't add migration guide for other more advanced features. Very few of customers will use that.